controllers/bsl.go

```diff
@@ -93,11 +93,7 @@ func (r *DPAReconciler) ValidateBackupStorageLocations(dpa oadpv1alpha1.DataProt
 	return true, nil
 }
 
-func (r *DPAReconciler) ReconcileBackupStorageLocations(log logr.Logger) (bool, error) {
-	dpa := oadpv1alpha1.DataProtectionApplication{}
-	if err := r.Get(r.Context, r.NamespacedName, &dpa); err != nil {
-		return false, err
-	}
+func (r *DPAReconciler) ReconcileBackupStorageLocations(log logr.Logger, dpa *oadpv1alpha1.DataProtectionApplication) (bool, error) {
```

Comment:
Instead of adding one more parameter to each function, is not better to add dpa here https://github.com/openshift/oadp-operator/blob/master/controllers/dpa_controller.go#L48-L55 ?

Reply:
@mateusoliveira43 do you mean to bind the DPAReconciler with the dpa something like this and use it everywhere then?

`type DPAReconciler struct {
	client.Client
	Scheme         *runtime.Scheme
	Log            logr.Logger
	Context        context.Context
	NamespacedName types.NamespacedName
	EventRecorder  record.EventRecorder
	dpa 		   *oadpv1alpha1.DataProtectionApplication
}`  ?

Reply:
yes :smile:

Reply:
Nice idea! Updated! @mateusoliveira43